### PR TITLE
fix(github): bound the community-followups comment scan

### PR DESCRIPTION
**Pull request by arnavpopley**
Part of #5121 (covers one of the read-only investigation tools; the remaining write/mutation tools are not addressed by this PR, so it should not auto-close that issue)

#### Describe the changes you have made in this PR -

Found while live-verifying `summarize_community_followups` against a real GitHub Copilot MCP session targeting `Tracer-Cloud/opensre`. The tool queries `/repos/{owner}/{repo}/issues/comments`, which returns every comment across the **whole repository** (not scoped to one issue). `GitHubRestClient.paginate()` follows every `Link`-header page with no cap of any kind (`while url:`), so on an active repository with thousands of issues and comments this call ran to hundreds of pages. Reproduced live: the call was still running after 100+ seconds against `Tracer-Cloud/opensre` and had to be killed.

Two changes:

1. `GitHubRestClient.paginate()` now accepts `max_pages` (default 50 pages / 5,000 items at the default page size) and stops there, returning whatever was collected rather than raising. This is a shared safety net for every caller of `paginate()`, not just this one tool — the same unbounded-`while` pattern could bite any future endpoint that returns a repo-wide (rather than per-resource) listing.
2. `summarize_community_followups` adds a `since_days` parameter (default 30), passed as GitHub's native `since` query param plus `sort=updated&direction=desc`. This is also the semantically correct fix, not just a performance workaround: the tool's own stated purpose is "finding unanswered contributor questions" and "meeting agenda follow-ups" — recent activity, not a full-repository comment history scan.

### Demo/Screenshot for feature changes and bug fixes -

Before the fix, against `Tracer-Cloud/opensre` (real token, real repo):
```
$ time summarize_community_followups(owner="Tracer-Cloud", repo="opensre", per_page=20, github_token=token)
[killed after 100+ seconds, still running]
```

After the fix, same call with `since_days=7`:
```
$ time summarize_community_followups(owner="Tracer-Cloud", repo="opensre", per_page=100, since_days=7, github_token=token)
{"source": "github", "available": true, "unanswered_questions": [...real, current PR/issue comments...]}
real  0m5.6s
```

Regression test proof: reverted the fix locally (`git stash` on the two source files) and reran the two new tests — both failed with the exact symptom (`TypeError: unexpected keyword argument 'max_pages'`, and `since` missing from the request params), then restored the fix and confirmed green.

Local checks run: `make lint`, `make format-check`, `make typecheck` (all clean), `tests/integrations/test_github_client.py` + `tests/tools/test_github_community_followup_tool.py` (new file — no test coverage existed for this tool before) + the full `integrations/github/tools/` scoped suite per `.github/ci/test_scope_rules.py` (203 passed).

---

## Code Understanding and AI Usage

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**

Found this while live-verifying GitHub's read-only investigation tools end-to-end (real MCP session, real repo, real token) rather than against mocked responses — the call visibly hung well past what any reasonable tool-call timeout in an investigation loop would allow. I traced it to `GitHubRestClient.paginate()`'s unconditional `while url:` loop and confirmed via `/repos/{owner}/{repo}/issues/comments`'s actual API docs that it supports a `since` filter, which is both the fix and the semantically correct scope for what this tool is meant to do. I added the `max_pages` cap at the shared client level (rather than only in this one caller) because the underlying defect — no bound on pagination depth — is a property of the client, and any other endpoint that returns a repo-wide rather than per-resource listing would hit the same failure mode.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

### CODE REVIEW

File: integrations/github/tools/community_followup_tool/__init__.py [hunk 1]
```diff
@@ -67,18 +73,24 @@ def summarize_community_followups(
...
 ) -> dict[str, Any]:
     try:
-        normalized_comments = (
-            comments
-            if comments is not None
-            else GitHubRestClient(github_token).paginate(
+        if comments is not None:
+            normalized_comments = comments
+        else:
+            since = datetime.now(UTC) - timedelta(days=max(1, since_days))
```
[the remark this answers is not in the document]
Comment (arnavpopley) [reply integrations/github/tools/community_followup_tool/__init__.py]: Resolved

**Comment by arnavpopley** [comment]
@greptile review
